e2e/scenario_helpers_test.go

```diff
@@ -153,3 +159,73 @@ func getExpectedPackageVersions(packageName, distro, release string) []string {
 	}
 	return expectedVersions
 }
+
+func getCustomScriptExtensionStatus(ctx context.Context, t *testing.T, resourceGroupName, vmssName string) error {
+	pager := config.Azure.VMSSVM.NewListPager(resourceGroupName, vmssName, nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get VMSS instances: %v", err)
+		}
+
+		for _, vmInstance := range page.Value {
+			instanceViewResp, err := config.Azure.VMSSVM.GetInstanceView(ctx, resourceGroupName, vmssName, *vmInstance.InstanceID, nil)
+			if err != nil {
+				t.Logf("failed to get instance view for VM %s: %v", *vmInstance.InstanceID, err)
+				continue
+			}
+			for _, extension := range instanceViewResp.Extensions {
+				for _, status := range extension.Statuses {
+					resp, err := parseLinuxCSEMessage(t, *status)
+					if err != nil {
+						return fmt.Errorf("Parse CSE message with error, error code: %s, raw message: %s", err.Code, *status.Message)
+					}
+					if !strings.EqualFold(resp.ExitCode, "0") {
+						return fmt.Errorf("vmssCSE %s, output=%s, error=%s", resp.ExitCode, resp.Output, resp.Error)
+					} else {
+						t.Logf("CSE completed successfully with exit code 0")
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func parseLinuxCSEMessage(t *testing.T, status armcompute.InstanceViewStatus) (*datamodel.CSEStatus, *datamodel.CSEStatusParsingError) {
+	if status.Code == nil || status.Message == nil {
+		t.Logf("No valid Status code or Message provided from cse extension")
```

Comment:
string duplicate, maybe create errorMsg := "No valid Status code or Message provided from cse extension"

Reply:
I usually try to avoid logging if an error is returned. It generally just adds noise.

Reply:
fixed, remove duplicate log message and just returned error